datafusion/sql/src/statement.rs

```diff
@@ -625,8 +626,136 @@ impl<'a, S: ContextProvider> SqlToRel<'a, S> {
                 });
                 Ok(LogicalPlan::Statement(statement))
             }
+            Statement::CreateFunction {
+                or_replace,
+                temporary,
+                name,
+                args,
+                return_type,
+                params,
+            } => {
+                let return_type = match return_type {
+                    Some(t) => Some(self.convert_data_type(&t)?),
+                    None => None,
+                };
+                let mut planner_context = PlannerContext::new();
+                let empty_schema = &DFSchema::empty();
+
+                let args = match args {
+                    Some(function_args) => {
+                        let function_args = function_args
+                            .into_iter()
+                            .map(|arg| {
+                                let data_type = self.convert_data_type(&arg.data_type)?;
+
+                                let default_expr = match arg.default_expr {
+                                    Some(expr) => Some(self.sql_to_expr(
+                                        expr,
+                                        empty_schema,
+                                        &mut planner_context,
+                                    )?),
+                                    None => None,
+                                };
+                                Ok(OperateFunctionArg {
+                                    name: arg.name,
+                                    default_expr,
+                                    data_type,
+                                })
+                            })
+                            .collect::<Result<Vec<OperateFunctionArg>>>();
+                        Some(function_args?)
+                    }
+                    None => None,
+                };
+
+                // Not sure if this is correct way to generate name
+                // postgresql function definition may have schema part as well
+                // datafusion at the moment does lookup based on given string
+                // `schema_name.function_name` will work even if there is no `schema_name`
+                let name: String = name
+                    .0
+                    .into_iter()
+                    .map(|i| i.value)
+                    .collect::<Vec<String>>()
+                    .join(".");
```

Comment:
I recommend in this case that we throw an error if someone tries to create a qualified function, and if it is important to support qualified function names in the future we can revisit the issue then

So
```
CREATE FUNCTION foo
```
would work

but
```
CREATE FUNCTION my.foo ... 
```

would error (not yet implemented error)



This is the corresponding code for table references https://github.com/apache/arrow-datafusion/blob/dd4263f843e093c807d63edf73a571b1ba2669b5/datafusion/common/src/column.rs#L71-L99

 is https://docs.rs/datafusion/latest/datafusion/sql/enum.TableReference.html 

But at the moment DataFusion doesn't put functions in a schema or catalog, it simply puts them some global namespace.